haystack/components/websearch/serpex.py

```diff
@@ -0,0 +1,207 @@
+            response.raise_for_status()
+            return response
+
+        self._make_request = make_request
+
+    def __del__(self):
+        """
+        Clean up resources when the component is deleted.
+
+        Closes the HTTP client to prevent resource leaks.
+        """
+        try:
+            if hasattr(self, "_client"):
+                self._client.close()
+        except Exception:
+            pass
+
+    def to_dict(self) -> dict[str, Any]:
+        """
+        Serializes the component to a dictionary.
+
+        :returns: Dictionary with serialized data.
+        """
+        return default_to_dict(
+            self,
+            api_key=self.api_key.to_dict(),
+            engine=self.engine,
+            num_results=self.num_results,
+            timeout=self.timeout,
+            retry_attempts=self.retry_attempts,
+        )
+
+    @classmethod
+    def from_dict(cls, data: dict[str, Any]) -> "SerpexWebSearch":
+        """
+        Deserializes the component from a dictionary.
+
+        :param data: Dictionary to deserialize from.
+        :returns: Deserialized component.
+        """
+        deserialize_secrets_inplace(data["init_parameters"], keys=["api_key"])
+        return default_from_dict(cls, data)
+
+    @component.output_types(documents=list[Document])
+    def run(
+        self,
+        query: str,
+        engine: Optional[Literal["auto", "google", "bing", "duckduckgo", "brave", "yahoo", "yandex"]] = None,
+        num_results: Optional[int] = None,
+        time_range: Optional[str] = None,
+    ) -> dict[str, list[Document]]:
+        """
+        Fetches web search results for the given query.
+
+        :param query: The search query string.
+        :param engine: Override the default search engine. If None, uses the engine from initialization.
+        :param num_results: Override the default number of results. If None, uses num_results from initialization.
+        :param time_range: Time range filter for results. Options: "all", "day", "week", "month", "year".
+                          Defaults to None (all time).
+        :returns: Dictionary containing a list of Document objects with search results.
+        """
+        documents: list[Document] = []
+
+        try:
+            # Prepare request parameters
+            params: dict[str, Any] = {
+                "q": query,
+                "engine": engine or self.engine,
+                "num": num_results or self.num_results,
+                "category": "web",
+            }
+
+            if time_range:
+                params["time_range"] = time_range
+
+            headers = {"Authorization": f"Bearer {self.api_key.resolve_value()}", "Content-Type": "application/json"}
+
+            # Make API request
+            response = self._make_request("https://api.serpex.dev/api/search", headers, params)
+            data = response.json()
+
+            # Parse search results (API returns 'results' not 'organic_results')
```

Comment:
Not sure if the part about the `organic_results` is relevant. If not let's drop it from the dev comment